File: apps/java/io/dustin/apps/board/api/qna/AnswerController.java
package io.dustin.apps.board.api.qna;

import io.dustin.apps.board.api.usecase.qna.answer.DeleteAnswerUseCase;
import io.dustin.apps.board.api.usecase.qna.answer.ModifyAnswerUseCase;
import io.dustin.apps.board.api.usecase.qna.answer.WriteAnswerUseCase;
import io.dustin.apps.board.domain.qna.answer.model.dto.AnswerDto;
import io.dustin.apps.common.exception.BadRequestParameterException;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.*;

import java.security.Principal;

@RestController
@RequestMapping("/api/v1/answer")
@RequiredArgsConstructor
public class AnswerController {

    private final WriteAnswerUseCase writeAnswerUseCase;
    private final ModifyAnswerUseCase modifyAnswerUseCase;
    private final DeleteAnswerUseCase deleteAnswerUseCase;

    //@PreAuthorize("isAuthenticated()")
    @PostMapping("/questions/{questionId}")
    public AnswerDto createAnswer(@PathVariable("questionId") Long questionId,
                                  @RequestBody AnswerDto answerDto) {
        if(answerDto.content() == null) {
            throw new BadRequestParameterException("댓글 내용은 필수항목입니다.");
        }
        return writeAnswerUseCase.execute(answerDto.adminId(), questionId, answerDto.content());
    }

    //@PreAuthorize("isAuthenticated()")
    @PatchMapping("/{answerId}")
    public AnswerDto modifyAnswer(@PathVariable("answerId") Long answerId,
                                  @RequestBody AnswerDto answerDto) {
        if(answerDto.content() == null) {
            throw new BadRequestParameterException("댓글 내용은 필수항목입니다.");
        }
        return modifyAnswerUseCase.execute(answerDto.adminId(), answerId, answerDto.content());
    }

    //@PreAuthorize("isAuthenticated()")
    @DeleteMapping("/{answerId}")
    public AnswerDto deleteAnswer(@PathVariable("answerId") Long answerId,
                                  @RequestBody AnswerDto answerDto) {
        return deleteAnswerUseCase.execute(answerDto.adminId(), answerId);
    }



}

File: apps/src/main/kotlin/io/dustin/apps/board/api/qna/AnswerController.kt
package io.dustin.apps.board.api.qna

import io.dustin.apps.board.api.qna.request.command.AnswerCreateCommand
import io.dustin.apps.board.api.qna.request.command.AnswerDeleteCommand
import io.dustin.apps.board.api.qna.request.command.AnswerModifyCommand
import io.dustin.apps.board.api.usecase.qna.answer.DeleteAnswerUseCase
import io.dustin.apps.board.api.usecase.qna.answer.ModifyAnswerUseCase
import io.dustin.apps.board.api.usecase.qna.answer.WriteAnswerUseCase
import io.dustin.apps.board.domain.qna.answer.model.dto.AnswerDto
import io.dustin.apps.common.code.CommonMessage
import io.dustin.apps.common.model.response.CommonResponse
import io.dustin.apps.common.model.response.ResultResponse
import io.swagger.v3.oas.annotations.Operation
import io.swagger.v3.oas.annotations.tags.Tag
import jakarta.validation.Valid
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.time.LocalDateTime


@RestController
@RequestMapping("/api/v1/answer")
@Tag(name = "1:1문의 답변 API", description = "1:1문의 답변 API 를 제공한다.")
class AnswerController (

    private val writeAnswerUseCase: WriteAnswerUseCase,
    private val modifyAnswerUseCase: ModifyAnswerUseCase,
    private val deleteAnswerUseCase: DeleteAnswerUseCase

) {
    /** =================== 1:1 문의에 대한 답변 endpoint =============== */

    @ResponseStatus(HttpStatus.OK)
    @PostMapping("/crate")
    @Operation(
        summary = "1:1 문의의 답변을 작성한다",
        description = "1:1 문의의 답변을 작성한다"
    )
    fun createAnswer(
        @RequestBody @Valid command: AnswerCreateCommand
    ): ResultResponse<AnswerDto> {
        return writeAnswerUseCase.execute(command)
    }

    @ResponseStatus(HttpStatus.OK)
    @PatchMapping("/modify")
    @Operation(
        summary = "1:1 문의의 답변을 수정한다",
        description = "1:1 문의의 답변을 수정한다"
    )
    fun modifyAnswer(
        @RequestBody @Valid command: AnswerModifyCommand
    ): ResultResponse<AnswerDto> {

        return modifyAnswerUseCase.execute(command)
    }

    @ResponseStatus(HttpStatus.OK)
    @DeleteMapping("/delete")
    @Operation(
        summary = "1:1 문의의 답변을 삭제한다",
        description = "1:1 문의의 답변을 삭제한다"
    )
    fun deleteAnswer(
        @RequestBody @Valid command: AnswerDeleteCommand
    ): CommonResponse {
        deleteAnswerUseCase.execute(command)
        return CommonResponse(
            code = HttpStatus.OK.value(),
            message = CommonMessage.SUCCESS.code,
            timestamp = LocalDateTime.now()
        )
    }
}
